### ISSUE TITLE
Islands component that accepts children

### ISSUE THREAD

**Issue by berlysia** [honojs/honox#47]
### What is the feature you are proposing?

Current islands component can not render `children` in the client side.

If we were achieve this, it means we have ["Donut Components"](https://frontendatscale.com/blog/donut-components/) pattern.

One of the reasons is that the child is not passed in [this part](https://github.com/honojs/honox/blob/a186c42bfd35c05954df614078e89fcb19c1263c/src/client/client.ts#L34-L36), but in addition to this, it seems necessary to modify the hydrate process when an island exists within an island.

**Comment by yusukebe** [comment]
Hi @berlysia 

This issue is resolved by #35. I'll release the new version, including the change, later.

**Comment by usualoma** [comment]
Hi @berlysia Thanks for the report!
cc @yusukebe 

#35 will partially fix this problem, but it does not work correctly when children is a function component. I have an idea to deal with this, so please give me a few days.

**Comment by usualoma** [comment]
Support children with server-side `Suspense` (streaming) is coming soon.

* Hydrate without waiting for the server side Suspense to resolve, and user can click button.
* Children are updated according to the server-side Suspense resolve.


https://github.com/honojs/honox/assets/30598/3103eefb-2e28-4a98-99de-da8c9dfe850c



```ts
const ChildContent3 = async () => {
  await new Promise((resolve) => setTimeout(resolve, 1000))
  throw new Error('Error in child content 3')
}

const ChildContent2 = async () => {
  await new Promise((resolve) => setTimeout(resolve, 1000))
  return (
    <>
      <p>Counter children 2</p>
      <ErrorBoundary fallback={<p>Something went wrong</p>}>
        <Suspense fallback="<p>Loading...</p>">
          <ChildContent3 />
        </Suspense>
      </ErrorBoundary>
    </>
  )
}

const ChildContent1 = async () => {
  await new Promise((resolve) => setTimeout(resolve, 1000))
  return (
    <>
      <p>Counter children 1</p>
      <Suspense fallback="<p>Loading...</p>">
        <ChildContent2 />
      </Suspense>
    </>
  )
}

const Children = async () => {
  return (
    <Suspense fallback="<p>Loading...</p>">
      <ChildContent1 />
    </Suspense>
  )
}

export default createRoute((c) => {
  const name = c.req.query('name') ?? 'no name'
  return c.render(
    <div class={className}>
      <h1>Hello, {name}!</h1>
      <Counter>
        <Children />
      </Counter>
    </div>,
    { title: name }
  )
})
```

**Comment by yusukebe** [comment]
@berlysia !

I've released v0.1.2, which supports that it can pass children to islands/client components (including Suspoense!) thanks to @usualoma . Could you try it?

**Comment by usualoma** [comment]
Oops, forgot to get the attribute values!
I'll fix that right away.
